### DISCUSSION TITLE
Suggestion: reducing variability in tournament scoring

### DISCUSSION THREAD

**Discussion by jacksondc** [yasserfarouk/scml#83] [Ideas]
Right now, if you run a tournament between two copies of the same agent (where all competitors and non-competitors are pursuing fully deterministic strategies), they won't receive the same score. The reason seems to be that, although the parameters are all carried over between worlds, the player who goes first in each negotiation is still selected randomly. If there's a way of keeping that selection consistent across runs—maybe fixing the random seed?—that might make it easier to get results.

**Comment by yasserfarouk** [comment]
You can fix who goes first by passing the following during world creation:

```python
world = SCMLOneshotWorld(
     ..., 
    mechanisms = {"negmas.sao.SAOMechanism": dict(avoid_ultimatum=False)}
)
```

I am not sure this will remove all sources of randomness. Let me know if it does
